enhancements/storage/manila-csi-driver-operator.md

```diff
@@ -0,0 +1,230 @@
+---
+title: manila-csi-driver-integration
+authors:
+  - "@mfedosin"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-11-22
+last-updated: 2019-11-22
+status: implementable
+---
+
+# Manila CSI driver integration
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This document describes Manila CSI driver integration to enable RWX volume access on OpenShift 4 on OpenStack.
+
+## Motivation
+
+Persistent Volumes, when running on OpenStack, are backed by Cinder, which doesn’t allow RWX access. Users have applications that require this access mode and want to run OpenShift on their OpenStack platforms. If we add Manila CSI support, users will be able to run more applications on OpenShift 4 on OpenStack. Also rolling upgrades for StatefulSets aren’t supported without RWX.
+
+### Goals
+
+- Bump the cloud-provider-openstack repo from upstream to include the latest changes for Manila CSI driver
+
+- Create a Manila CSI driver image and push it to Quay
+
+- Write Manila CSI driver operator that will manage the driver's lifecycle
+
+- Build Manila CSI driver operator image and push it to Quay
+
+- Add automatic testing
+
+### Non-Goals
+
+- Implement the CSI driver - it's already done in upstream
+
+- Implement CSI tooling (snapshotter, resizer, cloner). It is available in OpenShift since 4.3
+
+## Proposal
+
+Our main goal is to add RWX volume support in Openshift 4 on OpenStack. So we are going to use Manila through the CSI driver available in upstream as a part of cloud-provider-openstack repo.
+
+To maintain the lifecycle of the driver we want to implement an operator, that will handle all administrative tasks: deploy, restore, upgrade, healthchecks, etc.
+
+### User Stories
+
+#### Find a testing/development platform that supports Manila
+
+For testing and development we need a public OpenStack cloud with Manila support. So far MOC and PSI do not have this capability.
+The cloud should be OSP 13+ based, and comply with the reference architecture.
+There are two possibilities how we can accomplish this task:
+
+- Add Manila support to either MOC or PSI.
+
+- Find a completely new cloud, like next-gen RDO.
+
+Actions:
+
+- The first item on the list seems more likely. We need to contact the clouds admins and ask them to add Manila support.
+
+#### Update cloud-provider-openstack for OpenShift
```

Comment:
This should be noted in the _Risks_ section as well. The current state of openshift/cluster-api-provider-openstack (CAPO) is a blocker for this enhancement.

Reply:
Also a note. We will need to backport the Manilla features from v0.2 to v0.1 to support openshift CAPO